src/browser/components/tabbrowser/content/tabbrowser-js.patch

```diff
@@ -68,7 +68,21 @@ index ef857bd81f2cd7c163ecc74ac1cf81a0b63ce838..807eb0c493f15643412b05d8dad81d36
            if (!tabData.pinned) {
              this.unpinTab(tab);
            } else {
-@@ -3297,6 +3328,13 @@
+@@ -3283,6 +3314,13 @@
+             url = tabData.entries[activeIndex].url;
+           }
+ 
++          if(Services.prefs.getBoolPref("zen.pinned-tab-manager.restore-pinned-tabs-to-pinned-url", false) && tabData.pinned && tabData.zenPinnedUrl) {
```

Comment:
Same here, we could move it into the pin tabs manager and also add a lazy pref observer, to have less patches and not make a pref fetch every time